packages/bpk-component-barchart/src/BpkBarchart.js

```diff
@@ -45,17 +47,88 @@ const getClassName = cssModules(STYLES);
 const spacing = remToPx(spacingXs);
 const lineHeight = remToPx(lineHeightSm);
 
-const getMaxYValue = (dataPoints, yScaleDataKey, outlierPercentage) => {
+const getMaxYValue = (dataPoints: Function, outlierPercentage: ?number) => {
   const meanValue = dataPoints.reduce((d, t) => d + t, 0) / dataPoints.length;
   const maxYValue = Math.max(...dataPoints);
 
-  return outlierPercentage !== null
+  return outlierPercentage
     ? Math.min(maxYValue, meanValue * (outlierPercentage / 100) + meanValue)
     : maxYValue;
 };
 
-class BpkBarchart extends Component {
-  constructor(props) {
+type Props = {
+  data: any,
+  xScaleDataKey: string,
+  yScaleDataKey: string,
+  xAxisLabel: string,
+  yAxisLabel: string,
+  initialWidth: number,
+  initialHeight: number,
+  className: ?string,
+  leadingScrollIndicatorClassName: ?string,
+  trailingScrollIndicatorClassName: ?string,
+  outlierPercentage: ?number,
+  showGridlines: boolean,
+  xAxisMargin: number,
+  xAxisTickValue: Function,
+  xAxisTickOffset: number,
+  xAxisTickEvery: number,
+  yAxisMargin: number,
+  yAxisTickValue: Function,
+  yAxisNumTicks: ?number,
+  yAxisDomain: Array<?number>,
+  onBarClick: ?Function,
+  onBarHover: ?Function,
+  onBarFocus: ?Function,
+  getBarLabel: Function,
+  getBarSelection: Function,
+  BarComponent: Function,
```

Comment:
We typically try to avoid `Function` in favour of more specific syntax like `(argType) => returnType`, is there something more specific we can use here?

Reply:
Yeah these caused issues when I tried to type them exactly first time around :/ especially for BarComponent trying to type it correctly just constantly threw errors so not sure what to change it to here